### PR TITLE
Provide Software Sources for GPL and MPL software used by D2IQ.

### PR DESCRIPTION
**Pull request by orsenthil**
## Jira Ticket

After a recent scan, we determined that we need to link to software sources for GPL and MPL software used by D2IQ.

* https://jira.d2iq.com/browse/D2IQ-69446 

## Description of changes being made

* Added software-sources page and provided links to the software with licenses.
* Verified the license for the package and versions.

## Checklist
- [x] Change all affected versions, if applicable (e.g. 1.13, 2.0, 2.1).
- [x] Test all commands and procedures, if applicable.
- [x] Create your PR against `staging`, not `master`. 
- [x] Provide an estimated date for deploying the doc change. Note: Improvements or fixes can be merged ASAP. 
- [ ] Add [redirects](https://github.com/mesosphere/dcos-docs-site/wiki/Redirects) if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> ie `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/wiki/Contributing) for more information.

### CODE REVIEW

**Comment by masonse** [comment]
If you look at the Dispatch 1.2 Open Source Attributions (in our release notes), you'll see the following MPL-2.0 lines:
[open-source-attribution]$  cat index.md |grep MPL-2.0
|  github.com/armon | consul-api | v0.0.0-20180202201655-eb2c6b5be1b6 | [MPL-2.0](https://pkg.go.dev/github.com/armon/consul-api?tab=licenses) |
|  github.com/go-sql-driver | mysql | v1.4.1 | [MPL-2.0](https://pkg.go.dev/github.com/go-sql-driver/mysql?tab=licenses) |
|  github.com/hashicorp | errwrap | v1.0.0 | [MPL-2.0](https://pkg.go.dev/github.com/hashicorp/errwrap?tab=licenses) |
|  github.com/hashicorp | go-multierror | v1.0.0 | [MPL-2.0](https://pkg.go.dev/github.com/hashicorp/go-multierror?tab=licenses) |
|  github.com/hashicorp | go-uuid | v1.0.1 | [MPL-2.0](https://pkg.go.dev/github.com/hashicorp/go-uuid?tab=licenses) |
|  github.com/hashicorp | golang-lru | v0.5.0 | [MPL-2.0](https://pkg.go.dev/github.com/hashicorp/golang-lru?tab=licenses) |
|  github.com/hashicorp | golang-lru | v0.5.1 | [MPL-2.0](https://pkg.go.dev/github.com/hashicorp/golang-lru?tab=licenses) |
|  github.com/hashicorp | golang-lru | v0.5.3 | [MPL-2.0](https://pkg.go.dev/github.com/hashicorp/golang-lru?tab=licenses) |
|  github.com/hashicorp | golang-lru | v0.5.4 | [MPL-2.0](https://pkg.go.dev/github.com/hashicorp/golang-lru?tab=licenses) |
|  github.com/hashicorp | hcl | v1.0.0 | [MPL-2.0](https://pkg.go.dev/github.com/hashicorp/hcl?tab=licenses) |

**Comment by masonse** [comment]
@orsenthil Was the same software used to scan all of the software?  For Dispatch 1.2, @kevinmiller-conley used ShiftLeft Scan.  This is why you can see multiple versions of each package as different upstream dependencies may have used a different version of the same package (ie: golang etc).

File: pages/ksphere/konvoy/1.5/release-notes/software-sources/index.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+layout: layout.pug
+title: Software Sources
+menuWeight: 100
+excerpt: List of source software sources under GPL, and MPL used by D2IQ Konvoy
+render: mustache
+---
+
+The table below lists the Sources for the Free Software under GNU General Public License, and Mozilla Public License
```
Comment (dkoshkin) [inline pages/ksphere/konvoy/1.5/release-notes/software-sources/index.md:9]: Is this different from https://docs.d2iq.com/ksphere/konvoy/1.4/release-notes/open-source-attribution/?

**Comment by masonse** [comment]
Is there a legal requirement to list the software attributions twice?  Dispatch (and I'm assuming Kommander/Konvoy) already list all of their attributions on their docs page.

File: pages/ksphere/konvoy/1.5/release-notes/software-sources/index.md (hunk 1 above)
Comment (orsenthil) [reply pages/ksphere/konvoy/1.5/release-notes/software-sources/index.md:9]: Yeah,  different.

> "lists the Sources"

**Comment by orsenthil** [comment]
@masonse  - The important part of this change is *software sources* 

The listing of sources for direct dependencies for certain kinds of licenses is requested by legal and after a review that we provide the link to software sources.  One thing to note is important thing is - these modules will be present in go.mod in the (Dispatch and Konvoy) repo and they might be in Mozilla Public License.

Also, ref: https://github.com/mesosphere/dcos-docs-site/pull/3064#discussion_r448076923

**Comment by masonse** [comment]
@orsenthil Gotcha, the sources.  Could we just modify the existing markdown page(s) to include links to the sources for the modules? It might make it easier in the future to maintain one page per product that has all the OSS information on it.  As well as include this process in future OSS attribution page updates.

**Comment by cmastr** [comment]
> @orsenthil Gotcha, the sources. Could we just modify the existing markdown page(s) to include links to the sources for the modules? It might make it easier in the future to maintain one page per product that has all the OSS information on it. As well as include this process in future OSS attribution page updates.

This seems like a better idea to me as well vs. duplicating the same information in 2 places.

File: pages/ksphere/konvoy/1.5/release-notes/software-sources/index.md (hunk 1 above)
Comment (cmastr) [reply pages/ksphere/konvoy/1.5/release-notes/software-sources/index.md:9]: Only different in that these items need additional information.

**Comment by orsenthil** [comment]
Hi All,

I am going to add a new table under the OSS attribution page and provide the link to these sources.  This will address some comments here. 

We need to provide source links to certain software, specifically GPL and Mozilla Public License for the version that we are using. This is the obligation as per the license terms, and engineering was requested to make that change.  

This is not present in our existing page and is the new addition.

1) The OSS attribution page gives a link to both direct and indirect dependencies and includes software licenses under other licenses too, all of which are not the same.

2) I didn't want to confuse by adding additional columns to the table or invite more work for ourselves by trying to give all the sources, which is not required.

I hope this explains the motivation. To note, if any additional changes are requested after publishing the page, I am open to making further changes.   My secondary concern will the future automation, or easy of an update, which we can determine subsequently and we can iterate upon this.

HTH.

**Comment by orsenthil** [comment]
```
| github.com/hashicorp | golang-lru | v0.5.0 | MPL-2.0 |
| github.com/hashicorp | golang-lru | v0.5.1 | MPL-2.0 |
| github.com/hashicorp | golang-lru | v0.5.3 | MPL-2.0 |
| github.com/hashicorp | golang-lru | v0.5.4 | MPL-2.0 |
```
Not related to the current discussion. But here, ideally only `v0.5.4` should be listed as per : https://github.com/mesosphere/dispatch/blob/dev/go.mod#L32  . @masonse - do you know why the older versions are listed, should those be removed? (Again, not related to current discussion, but something to keep in mind).

**Comment by masonse** [comment]
@orsenthil They were listed because the tool run by KevinMC (ShiftLeft Scan) found all of them and when I asked Nathan about listing just the newest versus all versions, he said we should list all of them.

**Comment by orsenthil** [comment]
@masonse I looked at other versions listed for dispatch too and it seems to be listing multiple `go binary versions` etc.  It seems we need to identify the dependency output that is used by the binary (It matches `go.mod` and `go.sum` information) .  We could review and analyze that separately, as a separate task.

**Comment by orsenthil** [comment]
@cmastr  - I have modified the page to include the `## Software Sources` in the same Open source attribution page.

* Wrote HTML for DC/OS pages since it was HTML.
* Converted to Markdown for Dispatch and Konvoy.

Since we are adding this retroactively, I'd recommend that we go with the information and iterate later for any identified changes.

Please let me know if this looks good for you.

**Comment by cmastr** [comment]
thanks Senthil! This looks much better.

**Pull request merged**